oauth.go

```diff
@@ -868,7 +878,12 @@ func (rt *RoundTripper) RoundTrip(userRequest *http.Request) (*http.Response, er
 		allParams.Add(paramPairs[i].key, paramPairs[i].value)
 	}
 
-	baseString := rt.consumer.requestString(userRequest.Method, canonicalizeUrl(userRequest.URL), allParams)
+	serverURL := cloneURL(userRequest.URL)
```

Comment:
Something isn't right... only occasionally I'll get a nil pointer dereference on this line